src/pip/_internal/download.py

```diff
@@ -889,6 +890,19 @@ def unpack_url(
         write_delete_marker_file(location)
 
 
+def parse_content_disposition(content_disposition, default_filename):
+    # type: (str, str) -> str
+    """
+    Parse filename from content_disposition.
+    return default_filename if parsed result is empty
+    """
+    type, params = cgi.parse_header(content_disposition)
+    filename = params.get('filename')
+    if filename:
+        filename = os.path.basename(filename).split("\\")[-1].rstrip(".")
```

Comment:
Seeing this implementation, it might actually be better to add a _second_ function (called something like `sanitize_filename()`) and put the bulk of your unit tests in the unit tests of that function. Also, why is it necessary to split on `\` after calling `basename()`, and why splitting only with that character as opposed to others? Some code comments are probably warranted here explaining what we are protecting against. Lastly, what about left-stripping `.`, too?

Reply:
I added ``.split("\\")[-1]`` because ``os.path.basename("dir\\file")`` return "dir\\file" in linux.
Later I found out that "dir \\ file" is a valid filename in linux. so ``.split("\\")[-1]`` can be removed.

I added ``.rstrip(".")`` because os.path.basename("dir/..") return ".." and ".." is not a valid filename,
But only ".." can not lead to directory traversal. so ``.rstrip(".")``` can also be removed.

So only ``os.path.basename(filename)`` could fix this security issue already, so I decided to remove ``.split("\\")[-1]`` and ``.rstrip(".")``, is this ok?

Reply:
Isn't something like that still needed for Windows, though?

Reply:
In windows both os.path.basename("dir\\file") and os.path.basename("dir/file")  return "file". 
There is no other thing need to do in Windows, I think.